### ISSUE TITLE
Make form validation messages to use custom value for the %s param in message

### ISSUE THREAD

**Issue by bathpp** [ant-design/ant-design#14231]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently, the form validation message uses the id value from ```getFieldDecorator(id, options)``` as the %s param in default validation messages.

The ```field``` in rules is set to id by default, and there is no way to change it.
Should provide a way to use the label value/custom value.

### What does the proposed API look like?

```
rules:[
  {
    field: CUSTOM
  }
]
```
if field is not provided, use default as before

**Comment by afc163** [comment]
We can remove `%s` by default.

**Comment by zombieJ** [comment]
`async-validator` 里的 %s 是替换文字，不是对应的 field name。format 后如果有多余的 args 没有在文本中替换就添加到最后：https://github.com/yiminghe/async-validator/blob/master/src/util.js#L19
比如现在 `%s %s %s is required` 会变成 `file %s %s is required`。感觉可以剥出来，独立做个类似 `${field} is required` 然后替换掉，然后对以前的 %s 做一下兼容。
各位意下如何？

**Comment by syh2015** [comment]
> The same problem roasted in [Zhihu Answer](https://www.zhihu.com/question/33629737/answer/569954447).
> 
> ![image](https://user-images.githubusercontent.com/14918822/50942061-3fe9ab00-14c2-11e9-8517-f69570083c65.png)

这问题现在有解决了吗？

**Comment by zombieJ** [comment]
[Ant Design 4.0](#16911) 中，会变成更方便的模板形式：
```js
validateMessages={
  required: '${name} is required',
  // ...
}
```

**Comment by rohit-smpx** [comment]
> [Ant Design 4.0](#16911) 中，会变成更方便的模板形式：
> 
> ```js
> validateMessages={
>   required: '${name} is required',
>   // ...
> }
> ```

We can use custom message template in ant@4 but there is no way to provide additional properties like `label` to the message generator other than through rules. Which breaks types and also has some unintended side effects.
Related issue: https://github.com/react-component/field-form/issues/68

**Comment by piyushk96** [comment]
> > [Ant Design 4.0](#16911) 中，会变成更方便的模板形式：
> > ```js
> > validateMessages={
> >   required: '${name} is required',
> >   // ...
> > }
> > ```
> 
> We can use custom message template in ant@4 but there is no way to provide additional properties like `label` to the message generator other than through rules. Which breaks types and also has some unintended side effects.
> Related issue: [react-component/field-form#68](https://github.com/react-component/field-form/issues/68)

@zombieJ  @afc163 Any suggestions?

**Comment by zombieJ** [comment]
We tend user to use message if need customize error. But yes, if you have strong willing for this function. Please feel free to thumb up the https://github.com/react-component/field-form/issues/68 issue. We may reconsider if many developer need this function.

**Comment by Akimyou** [comment]
May be have may common times we want  use to the default valid error msg but need to custom the field name. Such as in `VeeValidate 2`.

```
  <input v-validate="'alpha'" data-vv-as="First Name" name="first_name" type="text">
```

How can i do same things in antd4?

**Comment by yoyo837** [comment]
https://ant.design/components/form/#validateMessages #23369 

Is this what you need?

**Comment by Akimyou** [comment]
> https://ant.design/components/form/#validateMessages #23369
> 
> Is this what you need?

How can i pass more params to `validateMessages` template?

**Comment by yoyo837** [comment]
You can't, unless you create a new feature request to improve it.

**Comment by Akimyou** [comment]
> You can't, unless you create a new feature request to improve it.

Hope can have this feature. Thanks.

**Comment by yoyo837** [comment]
You can create a new issue to request a new feature for that, where everyone can discuss how  to implement it or not.

**Comment by afc163** [comment]
closed in #23369
